### PR TITLE
copr: fix data race in the mppIterator

### LINKED ISSUE #39771: DATA RACE in the mppIterator
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
==================

WARNING: DATA RACE

[2022/12/08 22:45:28.632 +08:00] [WARN] [mpp.go:255] ["mpp dispatch meet io error"] [error="context canceled"] [timestamp=437914364445458432] [task=1]

Write at 0x00c00a8d26d8 by goroutine 725:

  runtime.racewrite()

      <autogenerated>:1 +0x24

  github.com/pingcap/tidb/store/copr.(*mppIterator).Close()

      store/copr/mpp.go:431 +0xc4

  github.com/pingcap/tidb/distsql.(*selectResult).Close()

      distsql/select_result.go:460 +0x36e

  github.com/pingcap/tidb/executor.(*MPPGather).Close()

      executor/mpp_gather.go:144 +0xdd

  github.com/pingcap/tidb/executor.(*recordSet).Close()

      executor/adapter.go:182 +0x5d

  github.com/pingcap/tidb/session.(*execStmtResult).Close()

      session/session.go:2401 +0x7e

  github.com/pingcap/tidb/testkit.(*TestKit).MustExecWithContext()

      testkit/testkit.go:135 +0x28f

  github.com/pingcap/tidb/testkit.(*TestKit).MustExec()

      testkit/testkit.go:125 +0xf7

  executor/tiflashtest/tiflashtest_test.TestBindingFromHistoryWithTiFlashBindable()

      executor/tiflashtest/tiflash_test.go:1275 +0x45c

  github.com/pingcap/tidb/domain.(*Domain).LoadSysVarCacheLoop()

      domain/domain.go:1360 +0xb9

  github.com/pingcap/tidb/session.BootstrapSession()

      session/session.go:3016 +0x78c

  github.com/pingcap/tidb/domain.(*Domain).GetSessionCache()

      domain/sysvar_cache.go:62 +0x6a

  github.com/pingcap/tidb/session.(*session).loadCommonGlobalVariablesIfNeeded()

      session/session.go:3277 +0x11e

  github.com/pingcap/tidb/session.(*session).ExecuteStmt()

      session/session.go:2125 +0x2fb

  github.com/pingcap/tidb/session.(*session).ExecuteInternal()

      session/session.go:1673 +0x4eb

  github.com/pingcap/tidb/domain.(*Domain).LoadPrivilegeLoop()

      domain/domain.go:1302 +0x143

  github.com/pingcap/tidb/session.BootstrapSession()

      session/session.go:3009 +0x72c

  github.com/pingcap/tidb/testkit.bootstrap()

      testkit/mockstore.go:84 +0x84

  github.com/pingcap/tidb/testkit.CreateMockStoreAndDomain()

      testkit/mockstore.go:69 +0xd0

  github.com/pingcap/tidb/testkit.CreateMockStore()

      testkit/mockstore.go:61 +0x3b2

  executor/tiflashtest/tiflashtest_test.TestBindingFromHistoryWithTiFlashBindable()

      executor/tiflashtest/tiflash_test.go:1262 +0x84

  testing.tRunner()

      GOROOT/src/testing/testing.go:1446 +0x216

  testing.(*T).Run.func1()

      GOROOT/src/testing/testing.go:1493 +0x47


Previous read at 0x00c00a8d26d8 by goroutine 1280:

  runtime.raceread()

      <autogenerated>:1 +0x24

  github.com/pingcap/tidb/store/copr.(*mppIterator).run()

      store/copr/mpp.go:175 +0x1dd

  github.com/pingcap/tidb/store/copr.(*MPPClient).DispatchMPPTasks.func1()

      store/copr/mpp.go:525 +0x58


Goroutine 725 (running) created at:

  testing.(*T).Run()

      GOROOT/src/testing/testing.go:1493 +0x75d

  testing.runTests.func1()

      GOROOT/src/testing/testing.go:1846 +0x99

  testing.tRunner()

      GOROOT/src/testing/testing.go:1446 +0x216

  testing.runTests()

      GOROOT/src/testing/testing.go:1844 +0x7ec

  testing.(*M).Run()

      GOROOT/src/testing/testing.go:1726 +0xa84

  go.uber.org/goleak.VerifyTestMain()

      external/org_uber_go_goleak/testmain.go:53 +0x70

  executor/tiflashtest/tiflashtest_test.TestMain()

      executor/tiflashtest/main_test.go:48 +0x61c

  main.main()

      bazel-out/k8-fastbuild/bin/executor/tiflashtest/tiflashtest_test_/testmain.go:174 +0x7c9


Goroutine 1280 (running) created at:

  github.com/pingcap/tidb/store/copr.(*MPPClient).DispatchMPPTasks()

      store/copr/mpp.go:525 +0x424

  github.com/pingcap/tidb/distsql.DispatchMPPTasks()

      distsql/distsql.go:45 +0x23a

  github.com/pingcap/tidb/executor.(*MPPGather).Open()

      executor/mpp_gather.go:127 +0x5d3

  github.com/pingcap/tidb/executor.(*ExecStmt).openExecutor()

      executor/adapter.go:1135 +0x109

  github.com/pingcap/tidb/executor.(*ExecStmt).Exec()

      executor/adapter.go:508 +0xbc4

  github.com/pingcap/tidb/session.runStmt()

      session/session.go:2346 +0x85c

  github.com/pingcap/tidb/session.(*session).ExecuteStmt()

      session/session.go:2210 +0x1266

  github.com/pingcap/tidb/testkit.(*TestKit).ExecWithContext()

      testkit/testkit.go:321 +0x84e

  github.com/pingcap/tidb/testkit.(*TestKit).MustExecWithContext()

      testkit/testkit.go:130 +0xb7

  github.com/pingcap/tidb/testkit.(*TestKit).MustExec()

      testkit/testkit.go:125 +0xf7

  executor/tiflashtest/tiflashtest_test.TestBindingFromHistoryWithTiFlashBindable()

      executor/tiflashtest/tiflash_test.go:1275 +0x45c

  github.com/pingcap/tidb/domain.(*Domain).LoadSysVarCacheLoop()

      domain/domain.go:1360 +0xb9

  github.com/pingcap/tidb/session.BootstrapSession()

      session/session.go:3016 +0x78c

  github.com/pingcap/tidb/domain.(*Domain).GetSessionCache()

      domain/sysvar_cache.go:62 +0x6a

  github.com/pingcap/tidb/session.(*session).loadCommonGlobalVariablesIfNeeded()

      session/session.go:3277 +0x11e

  github.com/pingcap/tidb/session.(*session).ExecuteStmt()

      session/session.go:2125 +0x2fb

  github.com/pingcap/tidb/session.(*session).ExecuteInternal()

      session/session.go:1673 +0x4eb

  github.com/pingcap/tidb/domain.(*Domain).LoadPrivilegeLoop()

      domain/domain.go:1302 +0x143

  github.com/pingcap/tidb/session.BootstrapSession()

      session/session.go:3009 +0x72c

  github.com/pingcap/tidb/testkit.bootstrap()

      testkit/mockstore.go:84 +0x84

  github.com/pingcap/tidb/testkit.CreateMockStoreAndDomain()

      testkit/mockstore.go:69 +0xd0

  github.com/pingcap/tidb/testkit.CreateMockStore()

      testkit/mockstore.go:61 +0x3b2

  executor/tiflashtest/tiflashtest_test.TestBindingFromHistoryWithTiFlashBindable()

      executor/tiflashtest/tiflash_test.go:1262 +0x84

  testing.tRunner()

      GOROOT/src/testing/testing.go:1446 +0x216

  testing.(*T).Run.func1()

      GOROOT/src/testing/testing.go:1493 +0x47

==================
```

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #39771

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
copr: fix data race in the mppIterator
```

### CODE REVIEW

File: store/copr/mpp.go [hunk 1]
```diff
@@ -446,7 +448,7 @@ func (m *mppIterator) Close() error {
 		close(m.finishCh)
 	}
 	m.cancelFunc()
-	m.wg.Wait()
```
Comment (hawkingrei) [inline store/copr/mpp.go:449]: We must not use two ```wg.wait() ``` parallelly.

**Review by LittleFall — APPROVED** [review]

**Review by guo-shaoge — APPROVED** [review]

**Pull request merged**